File: TypeScript/abc188_b/main.ts
async function readAllInput(): Promise<string[]> {
    const input = await new Response(Deno.stdin.readable).text();
    return input.trim().split('\n');
}

function parseLineToNumbers(line: string): number[] {
    return line.trim().split(' ').map(Number);
}

function solve(aArr: number[], bArr: number[]): string {
    let sum = 0;
    for (let i = 0; i < aArr.length; i++) {
        sum += aArr[i] * bArr[i];
    }

    if (sum == 0) {
        return "Yes";
    }
    return "No";
}

const lines = await readAllInput();

const _n = Number(lines[0]);
const aArr = parseLineToNumbers(lines[1]);
const bArr = parseLineToNumbers(lines[2]);

console.log(solve(aArr, bArr))

export{};

File: Python/abc188_b/main.py
def solve(n, a, b):
    sum = 0
    for i in range(n):
        sum = sum + a[i] * b[i]
    if sum == 0:
        return "Yes"
    else:
        return "No"


def get_input_int():
    return int(input())


def get_int_list():
    return list(map(int, input().split()))


def main():
    n = get_input_int()
    a = get_int_list()
    b = get_int_list()
    print(solve(n, a, b))


if __name__ == "__main__":
    main()
